src/android/app/src/main/java/org/citra/citra_emu/activities/EmulationActivity.kt

```diff
@@ -109,6 +109,16 @@ class EmulationActivity : AppCompatActivity() {
         isEmulationRunning = true
         instance = this
 
+        val game = try {
+            intent.extras?.let { extras ->
+                BundleCompat.getParcelable(extras, "game", Game::class.java)
+            } ?: throw IllegalStateException("Missing game data in intent extras")
+        } catch (e: Exception) {
+            throw IllegalStateException("Failed to retrieve game data: ${e.message}", e)
```

Comment:
What happens if this is thrown? Does the app crash? Wouldn't an error message be better?